internal/job/repository_job_run.go

```diff
@@ -0,0 +1,325 @@
+package job
+
+import (
+	"context"
+	"fmt"
+	"strings"
+
+	"github.com/hashicorp/boundary/internal/db"
+	"github.com/hashicorp/boundary/internal/db/common"
+	"github.com/hashicorp/boundary/internal/db/timestamp"
+	"github.com/hashicorp/boundary/internal/errors"
+	"github.com/hashicorp/boundary/internal/kms"
+	"github.com/hashicorp/boundary/internal/oplog"
+	"github.com/hashicorp/boundary/internal/types/scope"
+	"google.golang.org/protobuf/proto"
+)
+
+// FetchWork queries the job repository for available work and returns a JobRun
+// of a job that needs to be run, if there are not jobs that need to be run `nil, nil`
+// will be returned.
+//
+// • serverId is required and is the private_id of the server that will run the job.
+//
+// All options are ignored.
+func (r *Repository) FetchWork(ctx context.Context, serverId string, _ ...Option) (*JobRun, error) {
```

Comment:
Is a controller only allowed to run one job at a time?

Reply:
Also, the name `FetchWork` seems odd to me for two reasons. First, calling this method results in changes to the database and I wouldn't expect anything named "Fetch" to make changes. And second, "Work" is not used in any of the types in this package.

Reply:
No, since there is no denylisting currently being created I wanted to limit the work allocation to a single job to give all controllers a chance to get work.  I was considering making this a input or a similar `WithLimit` option but haven't yet.

I went around in circles on this, I didn't want to name it `GetJobRun` because that seems even more like its just getting a JobRun from the database.  I didn't want to name it `CreateJobRun` because we might not actually create a `JobRun` 

Lets chat Monday